clippy_lints/src/operators/mod.rs

```diff
@@ -116,6 +116,11 @@ declare_clippy_lint! {
     /// # let a = 0;
     /// a + 1;
     /// ```
+    ///
+    /// ### Noteworthy
+    ///
+    /// The `arithmetic_side_effects` lint is smarter with integer types, can deal with more corner
+    /// cases and is suitable for both numeric and custom types.
```

Comment:
Adding a reference ca be good, though based on this description, I'm a bit confused, why a user should pick this lint over the `arithmetic_side_effects`. It might be good to make the difference clearer, or is `arithmetic_side_effects` just better all around?

Reply:
There are a bunch of other reasons, in fact, `arithmetic_side_effects` was initially created to replace `integer_arithmetic`.

I just don't have the energy or motivation to search all my previous contributions to compile a list so I will simply remove the mention.

Reply:
I'm fine with being it mentioned, you could have a simple: `See also` in both lints that link to the other version. Just the way it was described there was a bit confusing. Removing it completely is also fine with me :).

I'd probably set the changelog entry to none in that case, since this PR has no user-facing changes, even though it's always awesome to have more tests :)